cpp/src/reader/table_query_executor.cc

```
@@ -85,6 +87,115 @@ int TableQueryExecutor::query(const std::string &table_name,
     return ret;
 }
 
-void TableQueryExecutor::destroy_query_data_set(ResultSet *qds) { delete qds; }
+int TableQueryExecutor::query_on_tree(
+    const std::vector<std::shared_ptr<IDeviceID>>& devices,
+    const std::vector<std::string>& tag_columns,
+    const std::vector<std::string>& field_columns, Filter* time_filter,
+    ResultSet*& ret_qds) {
+    common::PageArena pa;
+    pa.init(512, common::MOD_TSFILE_READER);
+    int ret = common::E_OK;
+    TsFileMeta* file_meta = tsfile_io_reader_->get_tsfile_meta();
+    std::vector<MetaIndexNode*> table_inodes;
+    for (auto const& device : devices) {
+        MetaIndexNode* table_inode;
+        if (RET_FAIL(file_meta->get_table_metaindex_node(
+                device->get_table_name(), table_inode))) {
+        };
+        table_inodes.push_back(table_inode);
```

Should return the failure？